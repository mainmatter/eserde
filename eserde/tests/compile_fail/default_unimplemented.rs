#[derive(eserde::Deserialize)]
struct Foo {
    #[serde(default)]
    field: NoDefault,
}

#[derive(eserde::Deserialize)]
struct NoDefault;

fn main() {}
