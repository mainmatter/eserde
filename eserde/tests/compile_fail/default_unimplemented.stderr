error[E0277]: the trait bound `NoDefault: Default` is not satisfied
 --> tests/compile_fail/default_unimplemented.rs:3:5
  |
3 |     #[serde(default)]
  |     ^ the trait `Default` is not implemented for `NoDefault`
  |
help: consider annotating `NoDefault` with `#[derive(Default)]`
  |
8 + #[derive(Default)]
9 | struct NoDefault;
  |

error[E0277]: the trait bound `NoDefault: Default` is not satisfied
 --> tests/compile_fail/default_unimplemented.rs:1:10
  |
1 | #[derive(eserde::Deserialize)]
  |          ^^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `NoDefault`
  |
  = note: this error originates in the derive macro `::eserde::_serde::Deserialize` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `NoDefault` with `#[derive(Default)]`
  |
8 + #[derive(Default)]
9 | struct NoDefault;
  |
